google/cloud/storage/internal/openssl_util.cc

```diff
@@ -125,19 +125,19 @@ std::string Base64Encode(std::uint8_t const* bytes, std::size_t bytes_size) {
 }
 }  // namespace
 
-std::string Base64Decode(std::string const& str) {
+std::vector<std::uint8_t> Base64Decode(std::string const& str) {
 #ifdef OPENSSL_IS_BORINGSSL
   std::size_t decoded_size;
   EVP_DecodedLength(&decoded_size, str.size());
-  std::string result(decoded_size, '\0');
-  EVP_DecodeBase64(
-      reinterpret_cast<std::uint8_t*>(&result[0]), &decoded_size, result.size(),
-      reinterpret_cast<std::uint8_t const*>(str.data()), str.size());
+  std::vector<std::uint8_t> result(decoded_size);
+  EVP_DecodeBase64(result.data(), &decoded_size, result.size(),
+                   reinterpret_cast<std::uint8_t const*>(str.data()),
```

Comment:
This is another one of those cases where I would suggest casting to `unsigned char* const` (because it is coming from a `char [const]*`) and letting the compiler check that is the same as `std::uint8_t const*`.

Reply:
Fixed.  nit: we are now changing two different things in this PR.